packages/vscode-extension/src/commands/ImportObjects.ts

```diff
@@ -0,0 +1,238 @@
+/*
+ ** Copyright (c) 2021 Oracle and/or its affiliates.  All rights reserved.
+ ** Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+ */
+import * as fs from 'fs';
+import * as path from 'path';
+import { Uri, window } from 'vscode';
+import CustomObjectService from '../service/ImportObjectService';
+import { ANSWERS, ERRORS, IMPORT_OBJECTS } from '../service/TranslationKeys';
+import { actionResultStatus, ApplicationConstants, InteractiveAnswersValidator, ProjectInfoService } from '../util/ExtensionUtil';
+import BaseAction from './BaseAction';
+
+const objectTypes: {
+	name: string;
+	value: { name: string; type: string; prefix: string; hasRelatedFiles: boolean; relatedFiles?: { type: string }[] };
+}[] = require('@oracle/suitecloud-cli/src/metadata/ObjectTypesMetadata');
+
+const COMMAND_NAME = 'importobjects';
+
+export default class ImportObjects extends BaseAction {
+	private customObjectService: CustomObjectService;
+
+	constructor() {
+		super(COMMAND_NAME);
+		this.customObjectService = new CustomObjectService(this.messageService, this.translationService);
+	}
+
+	protected init(uri?: Uri) {
+		super.init(uri);
+		this.customObjectService.setVsConsoleLogger(this.vsConsoleLogger);
+	}
+
+	protected async execute() {
+		if (!this.activeFile) {
+			// Already checked in validate
+			return;
+		}
+		const projectInfoService = new ProjectInfoService(this.getProjectFolderPath());
+
+		const destinationFolder = this.getDestinationFolder(this.activeFile);
+		if (!destinationFolder.startsWith(ApplicationConstants.FOLDERS.OBJECTS)) {
+			this.messageService.showErrorMessage(this.translationService.getMessage(IMPORT_OBJECTS.ERROR.INCORRECT_FOLDER));
+			return;
+		}
+
+		let appId: string | undefined;
+		if (projectInfoService.isSuiteAppProject()) {
+			appId = await this.promptAppId(projectInfoService);
+			if (appId == undefined) {
+				this.messageService.showInformationMessage(this.translationService.getMessage(IMPORT_OBJECTS.PROCESS_CANCELED));
+				return;
+			}
+			if (appId == '') {
+				appId = undefined;
+			}
+		}
+
+		const selectedObjectTypes = await this.promptObjectTypes();
+
+		if (!selectedObjectTypes) {
+			this.messageService.showInformationMessage(this.translationService.getMessage(IMPORT_OBJECTS.PROCESS_CANCELED));
+			return;
+		}
+
+		const scriptId = await this.promptSelectedScriptId();
```

Comment:
Variable name could be scriptIdToFilterBy or scriptIdFilter or filterByScriptId. Method name could be promptFilterByScripIdQuestion.